docs/rest.md

```diff
@@ -0,0 +1,623 @@
+background operation.
+
+It's also possible to provide the package as the entire body of a `POST` (not a
+multipart request). In this case the header `X-Allow-Unsigned` may be used to
+allow sideloading unsigned packages.
+
+### PUT
+
+* Description: change configuration for active packages. It is an error to
+  attempt to change configuration for non-active packages; if a configuration
+  change is requested for a package that is not active in the system the whole
+  command is aborted even if other packages that are active are specified in
+  the same command.
+* Authorization: trusted
+* Operation: sync
+* Return: configuration for all listed packages
+
+The request body is expected to be a JSON object with keys being the qualified
+package name(s) to configure, and the values will be passed to the configure
+hooks of the packages. The background operation information will similarly
+list individual statuses of the configuration changes.
+
+#### Sample input:
+
+```javascript
+{
+   "dd.canonical": "some-option: 42",
+   "hello-world.canonical": "greeting: Hi"
+}
+```
+
+#### Sample result:
+
+```javascript
+{
+   "dd.canonical": "some-option: 42\nsome-other-option: true",
+   "hello-world.canonical": "greeting: Hi\nheader: false"
+}
+```
+
+## /1.0/packages/[name]
+### GET
+
+* Description: Details for a  package
+* Authorization: trusted
+* Operation: sync
+* Return: package details (as in `/1.0/packages/`)
+
+
+### POST
+
+* Description: Install, update, remove, purge, activate, deactivate, or
+  rollback the package
+* Authorization: trusted
+* Operation: async
+* Return: background operation or standard error
+
+#### Sample input
+
+```javascript
+{
+ "action": "install"
+}
+```
+
+#### Fields in the input object
+
+field      | ignored except in action | description
+-----------|-------------------|------------
+`action`   |                   | Required; a string, one of `install`, `update`, `remove`, `purge`, `activate`, `deactivate`, or `rollback`.
+`leave_old`| `install` `update` `remove` | A boolean, default is false (do not leave old packages around). Equivalent to commandline's `--no-gc`.
+`config`   | `install` | An object, passed to config after installation. See `.../config`. If missing, config isn't called.
+
+#### A note on licenses
+
+The output field in the operation object for installation may specify that an
+additional step is needed, to confirm the user accepts the license specified
+in the package that's installing. Like so:
+
+```javascript
+{
+ "resource": "/1.0/operations/xyzzy",
+// ... other operation fields ...,
+ "output": {
+   "license_ack_needed": true,
+   "license_text": "Long license text\n\nMay include newlines etc.",
+ }
+}
+```
+
+The client must then present this license text to the user and ask for their
+acceptance. The operation will not complete until the client accepts (or
+declines) the license, by POSTing the appropriate response to the
```

Comment:
We're talking about this workflow in the other branch, so I won't raise it here again.